crates/cheatcodes/spec/src/vm.rs

```diff
@@ -102,8 +102,6 @@ interface Vm {
         uint64 gasLimit;
         /// The total gas used.
         uint64 gasTotalUsed;
-        /// DEPRECATED: The amount of gas used for memory expansion. Ref: <https://github.com/foundry-rs/foundry/pull/7934#pullrequestreview-2069236939>
-        uint64 gasMemoryUsed;
```

Comment:
Given that we are conclusively not adding this back I think this PR will be a good time to remove the field all together cc @DaniPopes

Reply:
I think this would be a breaking change for users relying on `gasRefunded`/`gasRemaining` values due to changed abi-encoding of the struct?

Reply:
actually I think this would be a breaking change for all current users of `lastCallGas` because abi-decoding would just fail

Reply:
Good call, this is a breaking change that is not required to be made in this PR. I can revert this change

Reply:
Decided to revert this change, it is not carried over into #8945 or #8952